logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,27 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
 
 module LogStash
+
+  DEFAULT_SETTINGS = {
+    "node.name" => Socket.gethostname,
```

Comment:
making all those strings constants would introduce another block of code with just constant definitions, create a level of indirection and make things harder to read (IMO).

This PR introduces settings that can be either expressed as a single string "pipeline.batch.size" or in hierarchical fashion:

```yaml
pipeline:
  batch:
    size: 6
```

for examples see the included `settings.yml`